dashboard/lib/queries/user/enabled_experiments.rb

```diff
@@ -0,0 +1,43 @@
+# Query object for retrieving enabled experiment names for a given user.
+#
+# Provides methods to get the current user's enabled experiments and
+# any experiments enabled for the user's associated teachers.
+#
+# Usage:
+#   query = Queries::User::EnabledExperiments.new(user)
+#   query.call            # => enabled experiment names for the user
+#   query.from_teachers   # => unique experiment names from user's teachers
+module Queries
+  module User
+    class EnabledExperiments < Queries::Base
+      def initialize(user)
+        @user = user
+      end
+
+      # Returns the enabled experiment names for the user.
+      #
+      # @return [Array<String>] list of experiment names
+      def call
+        experiment_names
+      end
+
+      # Gets all unique experiment names enabled for the user's teachers.
+      #
+      # @return [Array<String>] unique experiment names from teachers
+      def from_teachers
```

Comment:
I think having another public method that can be invoked outside of `.call` is conflating two schools of thought. The one school of thought is that a service object (or query object in this case) should only callable via the `call` method (the `call` method is the only public method). The second school of thought is to move related methods to a service (query) object as a PORO, and allow multiple public methods.

In our case, I believe we are trying to standardize on the first school of thought, where this class would only have a single public method, which is `call`, and everything else in this class is private.

Reply:
Yeah, the big [LTI Query](https://github.com/code-dot-org/code-dot-org/blob/104c6478a69fedd9e4553cd796c2bee153bc9c42/dashboard/lib/queries/lti.rb) class was implemented incorrectly, before we understood that each query object should only do one thing. We later started doing it correctly with [LTI Services](https://github.com/code-dot-org/code-dot-org/tree/104c6478a69fedd9e4553cd796c2bee153bc9c42/dashboard/lib/services/lti).

I think moving `from_teachers` to it's own Query object is a good choice.